src/core/exec-invoke.c

```diff
@@ -5431,6 +5472,45 @@ int exec_invoke(
         if (r < 0)
                 return r;
 
+        if (context->private_bpf) {
+                _cleanup_close_ int fs_fd = -EBADF, mnt_fd = -EBADF;
+
+                fs_fd = fsopen("bpf", 0);
+                if (fs_fd < 0) {
+                        *exit_status = EX_OSERR;
+                        return fs_fd;
+                }
+
+                r = send_one_fd(token_fds[0], fs_fd, 0);
+                if (r < 0) {
+                        *exit_status = EX_IOERR;
+                        return r;
+                }
+
+                r = wait_for_terminate_and_check("(bpffs)", TAKE_PIDREF(bpftoken_pid).pid, 0);
+                if (r < 0) {
+                        *exit_status = EX_OSERR;
+                        return r;
+                }
+                if (r != EXIT_SUCCESS) {
+                        /* If something strange happened with the child, let's consider this fatal, too */
+                        *exit_status = EX_OSERR;
+                        return -EIO;
+                }
+
+                mnt_fd = receive_one_fd(token_fds[0], 0);
+                if (mnt_fd < 0) {
+                        *exit_status = EX_IOERR;
+                        return fs_fd;
+                }
+
+                r = move_mount(mnt_fd, "", AT_FDCWD, "/sys/fs/bpf", MOVE_MOUNT_F_EMPTY_PATH);
+                if (r < 0) {
+                        *exit_status = EX_OSERR;
+                        return r;
+                }
```

Comment:
This should still be done in setup_namespace(), you can pass the fd as part of `NamespaceParameters` to make it available in `setup_namespace()`